### PR TITLE
Collect to return a record type.

### PR DESCRIPTION
**Pull request by anderson-joyle**
Issues #915 and #920.

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/PowerFxEvaluationTests.cs [hunk 1]
```diff
@@ -208,6 +208,23 @@ private static (RecalcEngine engine, RecordValue parameters) MutationFunctionsTe
                 recordWithRecord3
             });
 
+            var rValueType = RecordType.Empty()
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/PowerFxEvaluationTests.cs:211]: >var [](http://example.com/codeflow?start=12&length=3)

This PR may provide an easy way to add these tests, without having to modify C# :  #926

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/CollectFunction.cs [hunk 2]
```diff
@@ -160,11 +160,7 @@ public override bool CheckTypes(TexlNode[] args, DType[] argTypes, IErrorContain
             Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
 
             var fValid = base.CheckTypes(args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
-            Contracts.Assert(returnType.IsTable);
-
-            // TASK: 75145: SPEC: what if the types align for arg0, but arg0 is not a name node? For example:
-            //      Collect( Filter(T,A<2), {A:10} )
-            // The current behavior is that Collect has no side effects for transient tables/collections.
+            //Contracts.Assert(returnType.IsTable);
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/CollectFunction.cs:163]: >//Contracts.Assert(returnType.IsTable); [](http://example.com/codeflow?start=12&length=39)

dead code?

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — APPROVED** [review]
:shipit:

**Pull request merged**